filesmanager/tests/test_filesmanager.py

```diff
@@ -0,0 +1,949 @@
+        """
+        base_path = "path/to/content"
+        name = "content"
+        self.xblock.content_paths.append(base_path)
+
+        new_base_path, new_name = self.xblock.generate_content_path(base_path, name)
+
+        self.assertEqual(f"{base_path} (1)", new_base_path)
+        self.assertEqual(f"{name} (1)", new_name)
+
+    def test_generate_content_path_with_multiple_existing_paths(self):
+        """
+        Check generate_content_path method with multiple existing paths.
+
+        Expected result:
+            - The base path and name are returned with a suffix.
+        """
+        base_path = "path/to/content"
+        name = "content"
+        self.xblock.content_paths.extend([base_path, base_path])
+
+        new_base_path, new_name = self.xblock.generate_content_path(base_path, name)
+
+        self.assertEqual(f"{base_path} (2)", new_base_path)
+        self.assertEqual(f"{name} (2)", new_name)
+
+    def test_create_directory(self):
+        """
+        Check create_directory method.
+
+        Expected result:
+            - The directory is created in the root directory.
+            - The content_paths is updated.
+        """
+        directory = {
+            "id": "directory1",
+            "parentId": "root",
+            "name": "Directory1",
+            "type": "directory",
+            "path": "Root/Directory1",
+            "metadata": {},
+            "children": [],
+        }
+        self.xblock.upload_file_to_directory = Mock()
+        target_directory = self.xblock.directories["children"]
+
+        self.xblock.create_directory(directory, target_directory)
+
+        self.xblock.upload_file_to_directory.assert_not_called()
+        self.assertIn(directory["path"], self.xblock.content_paths)
+        self.assertIn(directory, target_directory)
+
+    def test_create_directory_with_children(self):
+        """
+        Check create_directory method with children.
+
+        Expected result:
+            - The directory and its children are created.
+            - The content_paths is updated.
+        """
+        self.xblock.upload_file_to_directory = Mock()
+        directory = {
+            "id": "test-directory-id",
+            "name": "TestDirectory",
+            "type": "directory",
+            "path": "Root/TestDirectory",
+            "parentId": "root",
+            "metadata": {},
+            "children": [
+                {
+                    "id": "test-file-id",
+                    "name": "TestFile",
+                    "type": "file",
+                    "path": "Root/TestDirectory/TestFile",
+                    "metadata": {},
+                    "parentId": "test-directory-id",
+                }
+            ],
+        }
+
+        self.xblock.create_directory(directory, self.xblock.directories["children"])
+
+        self.xblock.upload_file_to_directory.assert_called_once()
+        self.assertIn(directory["path"], self.xblock.content_paths)
```

Comment:
can we check the new directory is added to the directory children?

Reply:
Done